src/internal/connector/exchange/service_iterators_test.go

```diff
@@ -0,0 +1,648 @@
+package exchange
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/common"
+	"github.com/alcionai/corso/src/internal/connector/exchange/api"
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/internal/connector/support"
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/internal/tester"
+	"github.com/alcionai/corso/src/pkg/account"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/path"
+	"github.com/alcionai/corso/src/pkg/selectors"
+)
+
+// ---------------------------------------------------------------------------
+// mocks
+// ---------------------------------------------------------------------------
+
+var _ addedAndRemovedItemIDsGetter = &mockGetter{}
+
+type (
+	mockGetter        map[string]mockGetterResults
+	mockGetterResults struct {
+		added    []string
+		removed  []string
+		newDelta api.DeltaUpdate
+		err      error
+	}
+)
+
+func (mg mockGetter) GetAddedAndRemovedItemIDs(
+	ctx context.Context,
+	userID, cID, prevDelta string,
+) (
+	[]string,
+	[]string,
+	api.DeltaUpdate,
+	error,
+) {
+	results, ok := mg[cID]
+	if !ok {
+		return nil, nil, api.DeltaUpdate{}, errors.New("mock not found for " + cID)
+	}
+
+	return results.added, results.removed, results.newDelta, results.err
+}
+
+var (
+	_ graph.ContainerResolver = &mockResolver{}
+	_ graph.CachedContainer   = &mockResolvedContainer{}
+)
+
+type (
```

Comment:
can you add a note for this that it could probably be merged with what's in the container resolver tests if we decide we want to do so

Reply:
I'll do you one better: they're now merged.